examples/hp_search_benchmarks/darts_cifar10_pytorch/model_def.py

```diff
@@ -19,12 +19,43 @@
     PyTorchTrialContext,
 )
 
+import determined as det
+
 from model import NetworkCIFAR as Network
 import utils
 
 Genotype = namedtuple("Genotype", "normal normal_concat reduce reduce_concat")
 
 
+def apply_constraints(hparams, num_params):
+    # Reject if num skip_connect >= 3 or <1 in either normal or reduce cell.
+    normal_count = 0
+    reduce_count = 0
+    for hp, val in hparams.items():
+        if val == "skip_connect":
+            if "normal" in hp:
+                normal_count += 1
+            elif "reduce" in hp:
+                reduce_count += 1
+    if normal_count >= 3 or reduce_count >= 3:
+        raise det.InvalidHP("too many skip_connect operations")
+    if normal_count == 0 or reduce_count == 0:
+        raise det.InvalidHP("too few skip_connect operations")
+    # Reject if fewer than 3 sep_conv_3x3 in normal cell.
+    normal_count = 0
+    for hp, val in hparams.items():
```

Comment:
Can we build this into the former for loop so we don't have to iterate through hparams.items() twice?